basics/uri.adoc

```diff
@@ -140,3 +323,50 @@ NOTE: Local URIs do not contain the authority (and begin with `/`)
 
 
 NOTE: Please also refer to https://devhints.io/semver[devhints.io/semver] for more examples of version strings
+
+
+== Best Practices
+
+A UUri contain UAuthority, UEntity, and UResource. <<Uri Formats>> defined earlier merely define how UAuthority, UEntity, and UResource are expressed (long string names, short-form string, or encoded byte array). Some formats are human readable, while others are not.
+
+uProtocol URI specifications complies with of https://datatracker.ietf.org/doc/html/rfc3986[RFC3986] that builds https://datatracker.ietf.org/doc/html/rfc1034[DNS concepts and facilities] to allow representation of UAuthorty as names or IP addresses. 
+
+UEntity name to id, and UResource name+instance+message to id, are declared in the protos and compiled into language specific code using  https://protobuf.dev/programming-guides/proto2/#options[Options]. Application layer (uP-L3) uEs (client and servers) then have available the all the metadata needed to populate UUri with both the ids and names for UEntity and UResource as well the application layer code can also convert to/from the name-based (long Uri Format) to ID based (Short or Micro Uri Formats) locally (no dynamic lookup required). uEs are then able to send to the link:../up-l1/README.adoc[uP-L1 uTransport layer] The fully populated UUri objects.
+
+link:../up-l2/README.adoc[Dispatchers] however do not have the id mapping metadata as they are not aware of the service definitions in the protos. As a result, dispatchers can only work with the URI information provided (names or ids). If a dispatcher is required to convert the Uri format, it must dynamically lookup (resolve through discovery) the name/ids. 
```

Comment:
A backoffice application that has many tenants, ie users, drivers of cars, will be subscribing to the same topic, say temperature, but from different cars according to what the driver specifies.
The driver specifies a VIN - this is the car to monitor the temperature.
There are many drivers.
Events come in to the backoffice application -> if the UAuthority does not contain the vin or there is no way to understand what device the event came from, correlating to the subscription request - the backoffice application has a big problem and can't support many drivers/cars.
In addition, to not leak data between drivers, it is very important that this correlation is a pure function!

Reply:
If VIN doesn't reach the BO directly, then we must make sure the BO can resolve that UUri form to the long form to add the VIN to Tamara's point.  That conversion may not be trivial.  
The vehicle would be the easiest place to do that conversion, at the cost of a longer message to the BO.

Reply:
Today we use local addresses within a device and then append authority in the streamer, I would expect we continue to do this moving forward